packages/astro/test/units/vite-plugin-astro-server/response.test.js

```diff
@@ -0,0 +1,63 @@
+import {
+	createBasicSettings,
+	createFs,
+	createRequestAndResponse,
+	defaultLogger,
+} from '../test-utils.js';
+import { fileURLToPath } from 'node:url';
+import { expect } from 'chai';
+import { createContainer } from '../../../dist/core/dev/container.js';
+import testAdapter from '../../test-adapter.js';
+
+const root = new URL('../../fixtures/api-routes/', import.meta.url);
+const fileSystem = {
+	'/src/pages/index.js': `export const GET = () => {
+		const headers = new Headers();
+		headers.append('x-single', 'single');
+		headers.append('x-triple', 'one');
+		headers.append('x-triple', 'two');
+		headers.append('x-triple', 'three');
+		headers.append('Set-cookie', 'hello');
+		headers.append('Set-Cookie', 'world');
+		return new Response(null, { headers });
+	}`,
```

Comment:
Copied over from the [`NodeApp` unit test](https://github.com/withastro/astro/blob/astro@4.2.7/packages/astro/test/units/app/headers.test.js#L61-L74).